TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
from scipy import stats
from matplotlib import pyplot as plt
import warnings
warnings.filterwarnings('ignore')

In [2]:
data = pd.read_csv("Costomer+OrderForm.csv")

In [3]:
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [4]:
data.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [5]:
data.dtypes

Phillippines    object
Indonesia       object
Malta           object
India           object
dtype: object

In [7]:
data['Phillippines'].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [8]:
data["Indonesia"].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [10]:
data["Malta"].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [11]:
data["India"].value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

Chi-square test

In [15]:
import scipy as sp
from scipy.stats import chi2_contingency

In [18]:
chistats = sp.stats.chi2_contingency([[271,267,269,280],[29,33,31,20]])
chistats

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

Here test t = 3.8589 and p_value = 0.277
Inference: As (p_value = 0.2771) > (α = 0.05); Accept Null Hypthesis i.e. Independence of categorical variables Thus, customer order forms defective % does not varies by centre